### PR TITLE
JAMES-2495 Update various JUnit supporting libraries

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by chibenwa** [comment]
We need a detailed analyses of what runs and what does not...

**Comment by chibenwa** [comment]
In utils-java8

```
[185b3a9faca2b00756684a4a558a294dc0e89d62] -------------------------------------------------------
[185b3a9faca2b00756684a4a558a294dc0e89d62]  T E S T S
[185b3a9faca2b00756684a4a558a294dc0e89d62] -------------------------------------------------------
[185b3a9faca2b00756684a4a558a294dc0e89d62] 
[185b3a9faca2b00756684a4a558a294dc0e89d62] Results :
[185b3a9faca2b00756684a4a558a294dc0e89d62] 
[185b3a9faca2b00756684a4a558a294dc0e89d62] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[185b3a9faca2b00756684a4a558a294dc0e89d62] 
```

**Comment by chibenwa** [comment]
We may want to try to update surefire, to see if this solves things...

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
test this please (no build output)

File: pom.xml [hunk 1]
```diff
@@ -2699,7 +2699,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>2.19.1</version>
+                    <version>2.22.0</version>
```
Comment (rouazana) [inline pom.xml]: so... does it work?

**Comment by trantienduchn** [comment]
same error last time
```
[bb8caef0ce0f1fb8e5ed088f08fd02212aebf73f] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project james-server-util-java8: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test failed: java.lang.ClassNotFoundException: org.apache.maven.plugin.surefire.log.api.ConsoleLogger -> [Help 1]
```

Since maven surefire 2.22.0 has supported native JUnit platform then we don't need to declare `<artifactId>junit-platform-surefire-provider</artifactId>` in pom.xml. And that artifact may is the cause of above error. When removing it from pom.xml, my maven test in `james-server-util-java8` goes well :)

Details:
https://github.com/apache/maven-surefire/pull/184
https://github.com/junit-team/junit5/issues/1367

A comment I saw from above issue conversations:
> When you update to Surefire 2.20.0, make sure to remove your junit-platform-surefire-providerplugin dependency declaration for the surefire plugin or you will keep getting this error.

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/1631

**Pull request closed without merging**